### ISSUE TITLE
Suggestion: provide a parseCSV method to read CSV files from SharePoint libraries using PnPJs (interested to contribute)

### ISSUE THREAD

**Issue by AsishP** [pnp/pnpjs#414]
### Category
- [X] Enhancement
- [ ] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [    -     ]

Please specify what version(s) of SharePoint you are targeting: [  SPO  ]

*If you are not using the latest release, please update and see if the issue is resolved before submitting an issue.*

### Expected / Desired Behavior / Question
Current there is no method to be able to read CSV from a SharePoint library. In a recent project, I did implement this logic but also consider this functionality would be helpful for greater audience. Please let me know if this makes sense and would be interested to contribute.

### Observed Behavior
No provided method right now.

Thank you for your feedback!

**Comment by patrick-rodgers** [comment]
Hi @AsishP - I understand parsing CSV files does come up a fair amount, but I don't think it is appropriate for adding into the library. There already exist [many open source options](https://www.npmjs.com/search?q=csv) and [a function you can grab with associated discussion on stack overflow](https://stackoverflow.com/questions/8493195/how-can-i-parse-a-csv-string-with-javascript-which-contains-comma-in-data).

One of the challenges with a library like this is deciding what to bring in and what to leave out. IMO csv parsing is a solved problem many times over and there isn't a ton of value add to bringing that functionality internal. We had a similar discussion around bringing in CAML builder.

What I think would be great is if you want to update the [docs page for items](https://github.com/pnp/pnpjs/blob/dev/packages/sp/docs/items.md) and include an example of loading a csv file folks could use that as a reference. How does that sound?

**Comment by AsishP** [comment]
Hi @patrick-rodgers, yeah agreed and completely understand the challenge to decide additional features. To your point, yes the stackoverflow article kind of outlines the complexity of parsing a CSV in its raw format. I went through something similar when implementing the same, so thought if we could include it in the library like a method such as the `getJSON()` method (thought parseCSV() suits better) would have been really helpful instead of everyone building the logic from scratch.

Regarding updating the docs, may I propose that it could add some snippet and then add a link to my Github repository with a code sample for the same. There is a fair bit of code in it (even though I have simplified it a lot) and keeping it in raw format without not being to try it or run it might not be that beneficial. suggestions?
